common/captcha/CaptchaSolverInterface.py

```diff
@@ -0,0 +1,90 @@
+from abc import ABC, abstractmethod
+import os
+import cv2
+import time
+
+
+class SolvedCaptcha:
+    """
+    A class interface for representing solved Captchas.
+    """
+
+    def __init__(self, image, text, answer=None, out_dir=None):
+        """
+        :param image: Unprocessed Captcha image
+        :param text: The text contained in the captcha image. (eg: 12+3=)
+        :param answer: The solved captcha's answer (eg: 15). In the 'jumbled text' captchas this is the same as 'text'.
+        :param out_dir: Where to write captchas to upon notify_correct() or notify_incorrect()
```

Comment:
Something to consider is whether we want to have different fields for success_dir and failure_dir, so we can (for example) not was disk space on successful captchas.

Reply:
The thought behind this was for if we wanted to train a better tesseract model or neural network in the future. 

In this case we would need lots of labelled ground truth, so I don’t really anticipate a situation where we’d only need incorrect captchas and not correct ones. 

It’s more likely we’d want to disable writing captchas entirely, for example if we were happy with the efficacy of the captcha solver - which in hindsight is something that should be permitted by leaving out_dir blank.

Reply:
Ahhh, that makes a lot of sense. Thanks for the explanation! I agree that we should either write both or neither based upon out_dir.